psamm/commands/completepath.py

```diff
@@ -0,0 +1,280 @@
+    function will generate an LP problem with implicit sinks for all compounds
+    except for ones involved in the gap-filling reactions. An L1 minimized FBA problem
+    is then solved and the result is a flux based synthesis pathway for a specific compound.
+    This returns a reaction ID, flux value, and reaction entry for each reaction in the metabolic
+    model.
+
+    Args:
+        model: :class:`psamm.datasource.native.NativeModel`.
+        model_rxns: A list of reaction IDs of all gap-filling and original model reactions.
+        model_cpds: A list of Compound Entries for each compound involved in any gap-filling reactions.
+        compound: A compound Id of a compound being unblocked by the gap-filling.
+    """
+
+    for compound_id in compound:
+        model = metabolic_model.copy()
+
+        for rxn in metabolic_model.reactions:
+            if rxn not in model_rxns:
+                model.remove_reaction(rxn)
+        reaction_id = 'Compound_Production'
+        reaction_ex = Reaction(Direction.Forward, {compound_id: -1})
+        model.database.set_reaction(reaction_id, reaction_ex)
+        model.add_reaction(reaction_id)
+
+        prob = solver.create_problem()
+
+        v = prob.namespace()
+
+
+        # We keep track of temporary constraints from the last optimization so
+        # that we can remove them during the next call to solve(). This is
+        # necessary because removing the constraint immediately after solving
+        # a model can invalidate the solution in some solvers.
+        temp_constr = []
+        remove_constr = []
+        # Define flux variables
+        for reaction_id in model.reactions:
+            lower, upper = model.limits[reaction_id]
+            v.define([reaction_id], lower=lower, upper=upper)
+        # Define constraints
+        massbalance_lhs = {compound: 0 for compound in model.compounds}
+        for spec, value in iteritems(model.matrix):
+                compound, reaction_id = spec
+                massbalance_lhs[compound] += v(reaction_id) * value
+        for compound, lhs in iteritems(massbalance_lhs):
+            if compound not in model_cpd_list:
+                prob.add_linear_constraints(lhs >= 0)
+            else:
+                prob.add_linear_constraints(lhs == 0)
+
+        obj_var = v(name='Compound_Production')
+        prob.set_objective(obj_var)
+        prob.solve(lp.ObjectiveSense.Maximize)
+        obj_flux = prob.result.get_value(obj_var)
+        prob.add_linear_constraints(obj_var >= obj_flux)
+
+        z = prob.namespace()
+        for reaction_id in model.reactions:
+            z.define([reaction_id], lower=0)
+
+        _z = z.set(model.reactions)
+        _v = v.set(model.reactions)
+
+        prob.add_linear_constraints(_z >= _v, _v >= -_z)
+
+        objective = z.expr(
+            (reaction_id, -1)
+            for reaction_id in model.reactions)
+        prob.set_objective(objective)
+        prob.solve(lp.ObjectiveSense.Maximize)
```

Comment:
I would do a minimize here instead of maximizing the negative zs. If you minimize, you can also create the objective simply with `objective = z.sum(model.reactions)`.